oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/xml/SysViewImportHandler.java

```diff
@@ -93,6 +95,39 @@ private void processNode(ImportState state, boolean start, boolean end)
         }
     }
 
+    //TODO replace with library method
+    private static @NotNull String suggestPrefix(String namespace) {
+        return "ns_" + UUID.randomUUID().toString().substring(0, 8);
+    }
+
+    private NameInfo createNameInfo(String svName) throws RepositoryException {
```

Comment:
I think this whole logic should be common across all JCR methods potentially creating new items (with potentially unregistered expanded names) such as 

- `Node.addNode(...)`
- `Node.setProperty(...)`
- `Session.move(...)`

Probably one needs to add a `SessionContext.getOrCreateOakPath(...)` similar to the `getOakPath(...)` which is called from all those methods. The implementation should delegate to a new method `PathMapper.getOrCreateOakPath(...)` which should automatically register all unknown prefixes.